lib/extensions/brakeman_fingerprint_patch.rb

```diff
@@ -0,0 +1,63 @@
+# Brakeman fingerprints account for the file location as part of the fingerprint
+# digest. The fingerprint uses the file's relative path, but assumes that all
+# files being scanned will be under the app_tree root directory, providing a
+# consistent file path regardless of the system it is run on.
+#
+# For engines, however this is not always the case. Engines can come from gems
+# and gems can be installed basically anywhere on the system depending on how
+# ruby is installed and configured, and even depending on which Ruby version
+# manager is used. Additionally, in CI gems are typically installed in a vendor
+# directory, and locally gems can be configured as git-based or path-based.
+# Because of this, the file path and the fingerprint can vary widely between
+# local dev and CI, causing problems when trying to ignore issues.
+#
+# This patch introduces a way to "remove" the leading path for files that reside
+# in engines. This removal provides a consistent file path for the fingerprint
+# method.
+#
+# NOTE: This patch only modifies what is necessary to make the
+# test:security:brakeman test suite work, namely fingerprint and the json
+# reporter (which leverages to_hash). It does not modify things such as the
+# the text reporter (CLI output) nor the interactive ignore.
+module BrakemanFingerprintPatch
+  def self.rails_engine_paths
+    @rails_engine_paths ||= ::Rails::Engine.subclasses.map { |e| e.root.to_s << "/" }
+  end
+
+  # Removes any leading engine paths if the file is an engine, and replaces with
+  # `(engine:<engine_name>)`
+  #
+  # NOTE: Ideally this code should use the in_engine_paths? method (that is
+  # patched in brakeman_excludes_patch.rb), however Brakeman::Warning doesn't
+  # have a reference to the app_tree where that method is defined, as warnings
+  # are standalone objects.
+  def file_string
+    engine_path = BrakemanFingerprintPatch.rails_engine_paths.detect { |p| self.file.absolute.start_with?(p) }
+    if engine_path
+      engine_name = File.basename(engine_path).sub(/-\h+$/, "").sub(/-(?:\d+\.)+\d+$/, "")
+      engine_relative = self.file.absolute.sub(engine_path, "")
+      "(engine:#{engine_name}) #{engine_relative}"
+    else
+      self.file.relative
```

Comment:
Do you have an example of what this returned when done on an engine?  I'm assuming this is the original code path that you're changing to engine:name relative path...

Reply:
Yes this is the original code path. 

I can post some examples

Reply:
@jrafanie I updated the comment within the code to include some examples.